### PR TITLE
initial commit qca port

### PR DESCRIPTION
**Pull request by gnome71**
This PR is to start a discussion about my work on a port on QCA. I have the problem, that I need Qt5 environment and how to find it on the users machine. I don't want to make a dependency to qt5 in the CONTROL file, because it is a huge download and very time consuming. 
Also on my machine I can just test it under Windows 10 and my Qt5 installation. So for now only msvc and msvc_64 is supported.
I'm open for any hints and critics - I'm still learning CMake.

Thanks for your help till now 
Alex

### CODE REVIEW

File: ports/qca/0001-fix-path-for-vcpkg.patch [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ endif()
+ 
+@@ -401,10 +401,10 @@ endif(DOXYGEN_FOUND)
+ include(CMakePackageConfigHelpers)
+ configure_package_config_file(
+   "${CMAKE_CURRENT_SOURCE_DIR}/QcaConfig.cmake.in"
+-  "${CMAKE_CURRENT_BINARY_DIR}/lib/cmake/${QCA_CONFIG_NAME_BASE}/${QCA_CONFIG_NAME_BASE}Config.cmake"
+-  INSTALL_DESTINATION ${QCA_LIBRARY_INSTALL_DIR}/cmake/${QCA_CONFIG_NAME_BASE}
++  "${CMAKE_CURRENT_BINARY_DIR}/share/cmake/${QCA_CONFIG_NAME_BASE}/${QCA_CONFIG_NAME_BASE}Config.cmake"
++  INSTALL_DESTINATION ${CMAKE_BINARY_DIR}/share/cmake/${QCA_CONFIG_NAME_BASE}
```
Comment (Orvid) [inline ports/qca/0001-fix-path-for-vcpkg.patch]: You want the path to be `/share/qca/` rather than `/share/cmake/`.

**Comment by ras0219-msft** [comment]
Unfortunately we really need to have this build against our managed version of Qt. This is to enable static linking scenarios where I have multiple components using Qt; if I want to static link `qca` and `qca_other` into my program, I need them to both be built against the same copy of Qt, otherwise, the symbols _from the two different qt versions_ will conflict and I won't be able to link my program.

I totally agree that the current `qt5` is a really big dependency -- we should do more work in the future to try to break it up into smaller pieces that can be modularly built. Also, remember that you only need to build it once and you can share it among all of your projects :). The layout is designed to be completely relocatable, so you CAN zip it up and move it between machines to reduce startup costs as well.

Other than that, your approach here looks pretty solid. I'll test it out once you fix the issue @Orvid picked out above.

**Comment by gnome71** [comment]
I changed the cmake file locations according to the issue @Orvid mentioned in his review. Also I added the build dependency to qt5 in the CONTROL file. So there is no need to search for qt on the users machine, nor has the user to set any env. variables.

Have a nice weekend!
Alex

**Comment by ras0219-msft** [comment]
Looks pretty good! There are four paths that catch my attention though:
- certs\
- mkspecs\
- debug\certs\
- debug\mkspecs\

`certs\` has `rootcerts.pem` in it, `mkspecs\` has a `.prf` file. Are these important and are they in the right place? I feel like the `pem` is probably not needed, but I don't know much about `.prf` files.

(Side note: the command `tree /f D:\src\vcpkg\packages\qca_x64-windows` is very nice to get an overview of the final package)

**Comment by gnome71** [comment]
According to:  https://www.openssl.org/docs/faq.html#USER16
it is up to developers or admins to maintain CAs.
Thats what the QCA Maintainers do with the rootcerts.pem file. According to the README the certs are converted from the Mozilla sources. But its quite outdated.
So my plan is to download a recent version maintained by the CURL developers and comment this in the portfile. So that everyone interested can follow the steps taken.

The portfile now builds the files mentioned by  @ras0219-msft in subdirs of PACKAGE/share/qca/ - where they fit better, I think.

**Comment by Orvid** [comment]
Alternatively, it might be viable to retrieve the cert-data from mozilla's main mirror and generate at build time. It would also mean that it can be versioned and we can bump the package version when it's updated: https://raw.githubusercontent.com/mozilla/gecko-dev/master/security/nss/lib/ckfw/builtins/certdata.txt

An even better solution might be to still do what I mentioned above but have it as a separate port so that it can be versioned independently. If going that route, it would likely be better to version directly off of nss: https://ftp.mozilla.org/pub/security/nss/releases/NSS_3_27_2_RTM/src/

**Comment by gnome71** [comment]
By now I'm not able to make an extra port for nss because it seems quite complicated for me. It uses the mozilla-build environment and I've not used that before.
By now I'm downloading [https://raw.githubusercontent.com/mozilla/gecko-dev/master/security/nss/lib/ckfw/builtins/certdata.txt](url) (like @Orvid mentioned) and process this file before building the library.

**Comment by ras0219-msft** [comment]
Thanks for working through this!

In the future, it would probably be best to separate out the `.pem` into it's own port, or perhaps even remove it entirely. Since the master CA list is somewhat volatile and security sensitive, it's more difficult to handle appropriately compared to normal software -- a user that wants to lock down all their software dependency versions will still probably want to update their CA list.

**Comment by gnome71** [comment]
I would prefer to remove it too. I will contact the library maintainers to clarify that and maybe remove it on the next version bump. 
Thanks for your guidance, it was a great experience for me to participate the first time on such a cool project. Keep up your good work!!

**Pull request merged**